src/argilla/client/feedback/dataset/remote/base.py

```diff
@@ -30,16 +28,20 @@
     import httpx
 
     from argilla.client.feedback.dataset.local import FeedbackDataset
+    from argilla.client.feedback.dataset.remote.dataset import RemoteFeedbackDataset
+    from argilla.client.feedback.dataset.remote.filtered import FilteredRemoteFeedbackDataset
     from argilla.client.feedback.schemas.types import AllowedFieldTypes, AllowedQuestionTypes
     from argilla.client.sdk.v1.datasets.models import FeedbackItemModel
     from argilla.client.workspaces import Workspace
 
 
 warnings.simplefilter("always", DeprecationWarning)
 
+T = TypeVar("T")
+
 
-class RemoteFeedbackRecords:
-    def __init__(self, dataset: "RemoteFeedbackDataset") -> None:
+class RemoteFeedbackRecordsBase:
```

Comment:
Do we need this records Base class? Could be possible to define the `FilteredRemoteFeedbackRecords` as specialization of the  `RemoteFeedbackRecords` class?

```python
class FilteredRemoteFeedbackRecords(RemoteFeedbackRecords): 
   def __init__(self, dataset, filters):
      super().__init__(dataset)
      self._filters = filters

   def __len__(self) -> int:
      # specific implementation using filters

   # overriding other methods to use filters
```

Reply:
We could do that, but we would need to overwrite every method in the base, which implies that there's not a clear base, just two classes doing different things, that's why I parametrised the base class, but I can turn it into a mixin if that makes more sense

Reply:
For me, the suffix `Base` refers to abstract classes, which contain a set of common methods and let the specific ones without implementation. A mixin is more like a way to extend behavior to different classes, right? So, yes if the base class is not defining all the methods, should be better a mixin